pystiche_papers/sanakoyeu_et_al_2018/_modules.py

```diff
@@ -184,3 +186,50 @@ def residual_block(channels: int) -> ResidualBlock:
     )
 
     return ResidualBlock(residual)
+
+
+class TransformerBlock(enc.SequentialEncoder):
+    r"""TransformerBlock from :cite:`SKL+2018`.
+
+    This block takes an image as input and produce a transformed image of the same size.
+
+    Args:
+        in_channels: Number of channels in the input. Defaults to ``3``.
+        kernel_size: Size of the convolving kernel. Defaults to ``10``.
+        stride: Stride of the convolution. Defaults to ``1``.
+        padding: Padding of the input. It can be either ``"valid"`` for no padding or
+            ``"same"`` for padding to preserve the size. Defaults to ``"same"``.
+        impl_params: If ``True``, use the parameters used in the reference
+            implementation of the original authors rather than what is described in
+            the paper.
+
+    If ``impl_params is True``, an :class:`~torch.nn.AvgPool2d` is used instead of a
+    :class:`~torch.nn.Conv2d` with :func:`~torch.nn.utils.weight_norm`.
+    """
+
+    def __init__(
+        self,
+        in_channels: int = 3,
+        kernel_size: Union[Tuple[int, int], int] = 10,
+        stride: Union[Tuple[int, int], int] = 1,
+        padding: str = "same",
+        impl_params: bool = True,
+    ):
+        kwargs = {
+            "kernel_size": kernel_size,
+            "stride": stride,
+            "padding": get_padding(padding, kernel_size),
+        }
+        module = (
+            nn.AvgPool2d(**kwargs)  # type: ignore[arg-type]
```

Comment:
Why do you need an `ignore[arg-type]` here? What is the error message? Is it maybe enough to declare `module: nn.Module = ...`?

Reply:
> pystiche_papers\sanakoyeu_et_al_2018\_modules.py: note: In member "__init__" of class "TransformerBlock":
pystiche_papers\sanakoyeu_et_al_2018\_modules.py:224: error: Argument 1 to "AvgPool2d" has incompatible type
"**Dict[str, Union[Tuple[int, int], int]]"; expected "bool"  [arg-type]
                nn.AvgPool2d(**kwargs)

At nn. Conv2d are similar error messages.

Reply:
Hm, OK I didn't think of that. In that case, I think it is OK as is.